analysis/simulation.py

```diff
@@ -36,38 +36,32 @@ def simulate_all(sequences, pfe_methods, network_models, font_directory, default
   """
   a_font_loader = font_loader.FontLoader(font_directory, default_font_id)
   result = dict()
-  for sequence, method in itertools.product(sequences, pfe_methods):
-
-    graphs = simulate_sequence(sequence, method, a_font_loader)
-    key = method.name()
-    results_for_key = result.get(key, list())
-    results_for_key.extend(totals_for_networks(graphs, network_models))
-    result[key] = results_for_key
-
+  for method in pfe_methods:
+    model_totals = dict()
+    for network_model in network_models:
+      totals = []
+      for sequence in sequences:
+        graphs = simulate_sequence(sequence, method, network_model, a_font_loader)
```

Comment:
Running the simulation per network model will multiply the total run time of the analysis. The good news is that most of the more costly methods to simulate (whole font, unicode range, optimal) don't and won't depend on network parameters. So you can eliminate most of the increased run time by adding a mechanism to have a method declare that it doesn't care about network parameters. For those methods you'd run the simulation once and then just copy relevant metrics to each network result.